core/api/service/author/requests/has_key.hpp

```diff
@@ -23,8 +24,11 @@ namespace kagome::api::author::request {
 
     outcome::result<Return> execute() override {
       OUTCOME_TRY(public_key, common::unhexWith0x(getParam<0>()));
-      return api_->hasKey(public_key,
-                          crypto::decodeKeyTypeFromStr(getParam<1>()));
+      if (auto key_type = crypto::KeyType::fromString(getParam<1>())) {
+        return api_->hasKey(public_key, *key_type);
+      } else {
+        return crypto::KeyTypeError::UNSUPPORTED_KEY_TYPE;
+      }
```

Comment:
I prefer more the other way for clarity (as you do in `bool KeyStore::prepare()`), but it is not an issue
```suggestion
      if (auto key_type = crypto::KeyType::fromString(getParam<1>())) {
        return api_->hasKey(public_key, *key_type);
      }
      return crypto::KeyTypeError::UNSUPPORTED_KEY_TYPE;
```